### PR TITLE
Update java.gradle for JDK 20

### PR DESCRIPTION
**Pull request by jgrassel**
Update `tasks.withType()` to be wise of Java 20's `javac` support removal for Java 7 as a source and compile target.

### CODE REVIEW

**Comment by jgrassel** [comment]
#build

**Comment by jgrassel** [comment]
#run-libby-bot #build

File: dev/wlp-gradle/java.gradle [hunk 1]
```diff
@@ -94,13 +94,18 @@ subprojects {
 
   
   tasks.withType(JavaCompile).configureEach({
+      int javaVersionNumVal = Integer.valueOf(JavaVersion.current().getMajorVersion())
```
Comment (jhanders34) [inline dev/wlp-gradle/java.gradle]:
```suggestion
      int javaVersionNumVal = Integer.parseInt(JavaVersion.current().getMajorVersion())
```

**Review by jhanders34 — CHANGES_REQUESTED** [review]

File: dev/wlp-gradle/java.gradle [hunk 2]
```diff
@@ -94,13 +94,18 @@ subprojects {
...
+      
+      if (('1.6'.equals(sourceCompatibility) || '1.7'.equals(sourceCompatibility)) && javaVersionNumVal >= 20) {
+          sourceCompatibility = "1.8"
+          targetCompatibility = "1.8"
+      } else if ('1.6'.equals(sourceCompatibility) && javaVersionNumVal >= 12) {
           sourceCompatibility = "1.7"
           targetCompatibility = "1.7"
-        }
       }
+
```
Comment (jhanders34) [inline dev/wlp-gradle/java.gradle]:
```suggestion

```

Comment (jhanders34) [reply dev/wlp-gradle/java.gradle]: The line has spaces.  Might as well make it an empty line with not spaces.  Not reason to have the line indented.

File: dev/wlp-gradle/java.gradle (hunk 1 above)
Comment (jgrassel) [reply dev/wlp-gradle/java.gradle]: updated

File: dev/wlp-gradle/java.gradle (hunk 2 above)
Comment (jgrassel) [reply dev/wlp-gradle/java.gradle]: updated

**Comment by jgrassel** [comment]
#run-libby-bot #build

**Review by jhanders34 — APPROVED** [review]

**Pull request merged**